src/tir/schedule/concrete_schedule.cc

```diff
@@ -500,6 +500,143 @@ Array<LoopRV> ConcreteScheduleNode::Split(const LoopRV& loop_rv,
   return CreateRV<LoopRV>(results);
 }
 
+Array<LoopRV> ConcreteScheduleNode::LoopPartition(const LoopRV& loop_rv,
+                                                  const Array<Optional<ExprRV>>& factor_rvs,
+                                                  bool preserve_unit_iters) {
+  class SymbolicShapeError : public ScheduleError {
+   public:
+    explicit SymbolicShapeError(IRModule mod, For loop) : mod_(mod), loop_(std::move(loop)) {}
+
+    String FastErrorString() const final {
+      return "ScheduleError: The min and extent values of the loop are required to be known at "
+             "compile time. However, dynamic shape has been detected.";
+    }
+
+    String DetailRenderTemplate() const final {
+      return "Detected dynamic shape in either min or extent of a loop {0}";
+    }
+
+    IRModule mod() const final { return mod_; }
+    Array<ObjectRef> LocationsOfInterest() const final { return {loop_}; }
+
+    IRModule mod_;
+    For loop_;
+  };
+
+  class NotSingleInferFactorError : public ScheduleError {
+   public:
+    explicit NotSingleInferFactorError(IRModule mod) : mod_(mod) {}
+
+    String FastErrorString() const final {
+      return "ScheduleError: only one factor can be specified as -1 or none";
+    }
+
+    String DetailRenderTemplate() const final {
+      return "Only one factor can be specified as -1 or none";
+    }
+
+    IRModule mod() const final { return mod_; }
+    Array<ObjectRef> LocationsOfInterest() const final { return {}; }
+
+    IRModule mod_;
+  };
+
+  class WrongFactorSumError : public ScheduleError {
+   public:
+    explicit WrongFactorSumError(IRModule mod, For loop) : mod_(mod), loop_(std::move(loop)) {}
+
+    String FastErrorString() const final {
+      return "ScheduleError: The sum of factors is larger than or equal to the extent of "
+             "loop";
+    }
+
+    String DetailRenderTemplate() const final {
+      return "The sum of factors is not larger than or equal to the extent of loop {0}";
+    }
+
+    IRModule mod() const final { return mod_; }
+    Array<ObjectRef> LocationsOfInterest() const final { return {loop_}; }
+
+    IRModule mod_;
+    For loop_;
+  };
+
+  class NonPositiveFactorError : public ScheduleError {
```

Comment:
Would be great to reuse the same error class as Split.

Reply:
Added changes to reuse the same error class for Split and Loop-Partition.